augur/refine.py

```diff
@@ -215,10 +218,13 @@ def run(args):
     import json
     tree_success = Phylo.write(T, tree_fname, 'newick', format_branch_length='%1.8f')
     print("updated tree written to",tree_fname, file=sys.stdout)
+
     if args.output_node_data:
         node_data_fname = args.output_node_data
-    else:
+    elif args.alignment:
```

Comment:
Ok, I'll make a PR with this change in `export`, and we can take a look and ensure it seems like it's going to work as intended, then we can modify `refine` here so that it doesn't output this if only node-names are added.

Reply:
I played around with this in `export`. It's simple to get branch lengths from the tree rather than branch_lengths node-data (but preferentially from branch_lengths node-data, if available). However, to test this, I'm inputting a 'raw' tree, then providing node-data (some is required) for only aa_muts (for example). 

So the deeper question is, if you `export` with a raw tree and some `ancestral`, or `trait` info also generated on a raw tree (so no internal node names match the tree), how should `export` handle this?

I played around and for V1, changed `recursively_decorate_tree_json_nextflu_schema` so that if `node['strain']` is `None` (this is an internal node, so has no name on a raw tree), it doesn't error, but displays a warning saying it can't find the node, and so metadata for this node won't be written. Essentially, the only node-data traits attached to the tree are those on the tips. 

Thanks to the wonderful raw tree I am using, this looks like a hot mess, but it works...:
![image](https://user-images.githubusercontent.com/14290674/57633761-4d17c300-75a4-11e9-947a-5f9df796491a.png)

By 'works' I mean that you can see different AA mutations at the tips, for example. Branches always seem to be just coloured by the first AA in the legend, I guess that's a default behaviour somewhere for if there's no other information for the branch? But assume this could be changed to default to grey?

For traits there ends up being no difference between just adding the trait from the metadata.

Is this the kind of behaviour we want? Or do we not want to support this?

(I tried to test this for V2 as well, but didn't realise it required reinstalling Auspice... so I've modified `export` to give me V2 JSONS without error, but I haven't yet tested them. Same theory applies as above.)

Reply:
Both the V1 exports that work in 'current' auspice and the V2 export do **NOT** work in V2 auspice.

Reply:
So no, it doesn't work for mutations, in V1 or V2. `--node-data` in `export` that can't be matched with nodes in the tree should cause an error (a reasonable one, unlike currently).

We could in theory still support `export` reading branch length from the tree instead of from a node-data file, in case they ran `refine` just to get node-names but didn't include the branch-length node-data JSON in `export`, but this is a question of how much we want to coddle users I suppose. If not, we should at least try to give a reasonable error probably.